### DISCUSSION TITLE
Switch from Emotion to StyleX(form Meta)

### DISCUSSION THREAD

**Discussion by StKostyantin** [chakra-ui/chakra-ui#8025] [Ideas]
### Description

Chakra uses the Emotion library. but it does not support ssr. As a suggestion to switch to a library created by Facebook called [StyleX](https://stylex-docusaurus.vercel.app/).

### Problem Statement/Justification

Emotion doesn't support ssr. and is inferior to modern libraries in terms of working with css in many other ways.

<img width="948" alt="image" src="https://github.com/chakra-ui/chakra-ui/assets/45784115/e44c88db-de92-4bf8-955e-4c15e513df97">

### Proposed Solution or API

Example of use

```
import stylex from '@stylexjs/stylex';

const styles = stylex.create({
  // base styles
  main: {
    width: '100%',
    maxWidth: 800,
    minHeight: 40,
  },
  // conditional styles
  active: {
    transform: 'scale(0.9)',
  },
});
```

```
import stylex from '@stylexjs/stylex';

const styles = stylex.create({ ... });
// Use styles
<div
  className={stylex(
    // Use base styles
    styles.main,
    // Use styles if a condition is true
    isActive && styles.active,
    // Use styles passed in as a prop
    props.xstyle,
  )}
/>;
```

### Additional Information
Docs: [https://stylex-docusaurus.vercel.app/docs/learn/](https://stylex-docusaurus.vercel.app/docs/learn/)
Github: [https://github.com/facebook/stylex](https://github.com/facebook/stylex)

**Comment by shannonhochkins** [comment]
This graph is wrong - Emotion for one, it supports everything you have ticked for StyleX

**Reply by sandrinjoy** [reply]
how is emotion atomic?

**Reply by shannonhochkins** [reply]
How is it not?

**Reply by sandrinjoy** [reply]
you can check how the classes are generated in emotion.  It creates one class for each style definition of an element, rather than keeping classes for each css property like tailwind, styleX , etc...

**Reply by shannonhochkins** [reply]
There is many ways to achieve atomic styles and classes with emotion, the downside is it's runtime unlike stylex

**Comment by jddeleon** [comment]
I think this is unlikely to happen, since the roadmap for Chakra V3 is already laid out, with the goal of using Panda as the new CSS-in-JS system.

https://www.adebayosegun.com/blog/the-future-of-chakra-ui

**Reply by strvbkzhn** [reply]
Don't you know when the Chakra V3 is gonna be released?

**Comment by nmn** [comment]
Panda has most of the same benefits as StyleX, it’s a good choice!
